Feature: Add a splash screen

Chrisg2000 opened the pull request:
Some users want the ability to display a splash screen when the bootloader is started (pyinstaller/pyinstaller#4354) 
This can be achieved by extending the bootloader and the possibility to define the necessary data in a spec file. 

This change implements the splash screen only for Windows, but provides a reasonable basis for extending it to other operating systems. 

A new file is added in the bootloader, which is responsible for an iOS-style splash screen. It provides a public (to the rest of the bootloader) API to call system dependent functions that display the splash screen.
Two functions are offered to display text on a user-definable area of the splash screen and to close the splash screen regardless of the bootloader's "unpacking state". This allows the splash screen to be continued to be used so that the Python program can initialize itself (perhaps process data, set up, etc.)

A new build target is defined in PyInstaller called "Splash", which packages the resources and settings for the splash screen into a binary file that can be attached to the application by EXE.

A Python module is defined, which the user can import into his program, which then connects to the bootloader/splash screen. This module then acts as RPC for the functions of the splash screen.

Chrisg2000 commented on PyInstaller/building/splash.py:
```diff
@@ -0,0 +1,291 @@
+                                          wnd_txt_clr=self.text_color,
+                                          transparency=transparency,
+                                          bitmap=bitmap_bits)
+        splash_res.write_to_file(self.name)
```
Okay, maybe one should check the line length in this code again and correct it if necessary (at least Travis/Flake8 is raising the alarm here)

Chrisg2000 commented on PyInstaller/hooks/pre_find_module_path/hook-pyi_splash.py:
```diff
@@ -0,0 +1,28 @@
+    module_dir = os.path.join(PACKAGEPATH, 'fake-modules')
+
+    api.search_dirs = [module_dir]
+    logger.info('Adding pyi_splash module to application dependencies')
```
Maybe before the module is redirected, check if a module with the same name is installed and if so, raise a SystemExit (or more descriptive Exception) ? Or rename the module to _pyi_splash? After all, it should not happen that the module is accidentally included although another one was meant.

Chrisg2000 replied:
I had initially implemented the module as a runtime hook, but then the following problems occurred:

- If it should be implemented as a runtime hook (i.e. in `loader/rthooks`), then modulegraph must find a module named `pyi_splash` during analysis, otherwise the runtime hook will not be included. So a fake module must be imported. During the boot process the runtime hook is then executed, which creates a module and injects it into `sys.modules` so that it can be imported. The runtime hook would then look somewhat like this:
  ```python3
  import sys

  is_win = sys.platform.startswith("win")

  MSG_FINISH_TYPE = b'c'
  MSG_UPDATE_TYPE = b'u'

  # A class of type module? Weird
  # type(sys) == type 'module'
  class _PYISplashModule(type(sys)):

      def __init__(self):
          import os  # Modules need to be imported where
                     # they are used, not globally
          # Only super() syntax works
          super(PYISplashModule, self).__init__('pyi_splash')

          self._initialized = False
          self._pipe_write_file = None

          ...

      # these methods cannot be found by e.g. pydoc
      def close(self):
          ...

      def update_text(self, msg):
          ...
  
  
  sys.modules['pyi_splash'] = _PYISplashModule()
  ```
  I just think that this code is more of a workaround than a reasonable solution and may have dangers that are not obvious. Also, a runtime hook will overwrite a module that is actually already implemented with the fake-module, so one can simplify it to the current solution. Unfortunately, one cannot pass the runtime hook via the target splash, because then there is the danger that the runtime hook is behind the user's script and therefore never executed.

- If it is to be implemented as a loader module (under `loader/`), it will always be executed, regardless of its position in the CArchive before the user's script (since modules are executed before scripts), but there are no non-builtin modules available at that time that pyi_splash needs (such as os, atexit). Again, a workaround can be used there, but it makes the code ugly and not maintainable.

So I chose this method.
The hook under `hooks/hook-pyi_splash.py` can be removed if Python 2 support should be completely removed, because it was only necessary for the dependencies under Python 2.

htgoebel commented on PyInstaller/building/splash.py:
```diff
@@ -0,0 +1,291 @@
+                    An optional alternative filename for the *.res file. If not specified,
+                    a name is generated.
+        """
+        super(Splash, self).__init__()
```
Please use old-style `Target.__init__(self)` to align with the other code.

htgoebel commented on PyInstaller/building/splash.py:
```diff
@@ -0,0 +1,291 @@
+                    a name is generated.
+        """
+        super(Splash, self).__init__()
+        from ..config import CONF
```
Please move to firt statement in the method

htgoebel commented on PyInstaller/building/splash.py:
```diff
@@ -0,0 +1,291 @@
+
+        # Make bitmap path relative to .spec file
+        if not os.path.isabs(bitmap):
+            bitmap = os.path.join(os.path.dirname(CONF['spec']), bitmap)
```
Three is `CONF['specpath']` already.

htgoebel commented on PyInstaller/building/splash.py:
```diff
@@ -0,0 +1,291 @@
+        if not os.path.isabs(bitmap):
+            bitmap = os.path.join(os.path.dirname(CONF['spec']), bitmap)
+        bitmap = os.path.normpath(bitmap)
+
```
Please remove this empty line.

htgoebel commented on PyInstaller/building/splash.py:
```diff
@@ -0,0 +1,291 @@
+            self.name = os.path.splitext(self.tocfilename)[0] + '.res'
+
+        self._check_txt_rect()
+
```
Please remove this empty line.

htgoebel commented on PyInstaller/building/splash.py:
```diff
@@ -0,0 +1,291 @@
+    def _check_txt_rect(self):
+        if len(self.text_rect) != 4:
+            raise ValueError("wnd_text_rect must be a tuple of four positive integers!")
+        if any([not (isinstance(a, int) and a > 0) for a in self.text_rect]):
```
You can leave of the `[` and `]`, which will make this into a generator and a bit easier to read. No need ot create a list first.

htgoebel commented on PyInstaller/building/splash.py:
```diff
@@ -0,0 +1,291 @@
+        if any([not (isinstance(a, int) and a > 0) for a in self.text_rect]):
+            raise ValueError("wnd_text_rect must be a tuple of four positive integers!")
+
+    def assemble(self):
```
I wonder what happens if one runs a .spec-file containing  `Splash()` on e.g. GNU/Linux? I assume we should check this case somewhere.

Chrisg2000 replied:
I left this check out because the build target Splash is largely independent from the operating system (except for a few small platform dependent adjustments to the data). So the data is passed on to the bootloader, no matter which operating system is running. If the bootloader supports the feature, it will display it, if not, they will be ignored. For this I have implemented the [message in the bootloader](https://github.com/Chrisg2000/pyinstaller/blob/splash-feature/bootloader/src/pyi_splash.c#L632 ). If this is implemented for other operating systems over time, then further changes can be made to the data.

Currently on a non-Windows system the resources would be attached but not used. Maybe a message can be displayed that the data is attached, but the target platform does not yet support this?

Chrisg2000 replied:
I have now added a[ warning message](https://github.com/Chrisg2000/pyinstaller/blob/splash-feature/PyInstaller/building/splash.py#L216) when the user is building for an unsupported platform. Nevertheless the data will still be attached.

htgoebel commented on bootloader/src/pyi_splash.c:
```diff
@@ -0,0 +1,113 @@
+ * The splash screen resources must be in the form of the SPLASH structure.
+ */
+
+/* TODO: use safe string functions */
```
Of ourse :-)

Chrisg2000 replied:
So i did that now too ;D

Chrisg2000 commented on bootloader/src/pyi_splash.c:
```diff
@@ -42,6 +48,537 @@ typedef struct _ipc_message_head {
+        return NULL;
+    }
+
+    /* Copy the bitmap from splash into the new bitmap */
```
I see no need to check the size, `SetDIBits` copies `splash->wnd_width` * `splash->wnd_height` pixels to `splash_bitmap`. The size of `splash->bitmap` is ignored, the return value indicates how many lines were read. But we should check if exactly as many lines were read as we wanted.

Chrisg2000 commented on bootloader/src/pyi_splash.c:
```diff
@@ -99,15 +646,53 @@ pyi_splash_launch(SPLASH *splash)
     /* This program receives the messages from the pipe so that they can be passed
      * on to the splash screen. Responses to messages are not supported,
      * so the commands must be checked for correctness in the client.*/
+    sprintf(_pipe_name, "\\\\.\\pipe\\pyi_splash_pipe_%d", getpid());
```
This is "special" Windows syntax. Under Windows, pipes are not files that are stored on disk, but extra objects that have their own URI.

Chrisg2000 commented on bootloader/src/pyi_splash.c:
```diff
@@ -0,0 +1,113 @@
+int
+pyi_splash_launch(SPLASH *splash)
+{
+    return 0;
```
I added a note to the commit message.

Chrisg2000 commented on bootloader/src/pyi_splash.c:
```diff
@@ -85,6 +622,16 @@ pyi_splash_get(ARCHIVE_STATUS *status)
+        splash);
+#else  /* if defined(_WIN32) */
+    VS("SPLASH: This bootloader does not support splash screen\n");
+    return -1;
```
Moved it into to first commit.

Chrisg2000 commented on PyInstaller/fake-modules/pyi_splash.py:
```diff
@@ -0,0 +1,241 @@
+is_win = sys.platform.startswith("win")
+is_py2 = sys.version_info[0] == 2
+MSG_FINISH_TYPE = b'c'
+MSG_UPDATE_TYPE = b'u'
```
Hope that comment makes it clearer

Chrisg2000 commented on bootloader/src/pyi_main.c:
```diff
@@ -123,6 +125,21 @@ pyi_main(int argc, char * argv[])
+    /* If the environment variable '_SPLASH' is set, the bootloader has just unpacked
+     * itself and it is a child process. In '_SPLASH' the pipe name for the pipe between
+     * the bootloader and the Python interpreter is stored. This environment variable is
+     * deleted as the python module pyi_splash is started/imported.
```
Indeed not, nevertheless I have rewritten the comment. The bootloader acts as a server that reads from the pipe. The pyi_splash module connects to the server as a client with write permission.

Chrisg2000 commented:
I have now changed a few things.
- The Python 2 support has now been removed.
- The files I added have a new copyright year. Should I also change the years in the changed files?
- Some comments have been improved/replaced.

I still have to write the tests, I didn't have much time for that yet. Also the documentation/changelog entries are still on the todolist. 

To the question if it makes sense to enable the splash screen in console mode: I would say, if the user defines it, then they should have it. Maybe someone wants to have a splash screen for a console application. The use cases can be versatile, e.g. [Blender 3D](https://www.blender.org/) also starts a console first and then opens the GUI with a splash screen.

Chrisg2000 commented on PyInstaller/fake-modules/pyi_splash.py:
```diff
@@ -0,0 +1,241 @@
+    :raises ConnectionError: If the OS fails to write to the pipe
+    """
+    # type checking/converting for writing to the pipe
+    if not isinstance(msg, (str, bytes, bytearray)):
```
The function is intended to update the text on the splash screen. I changed the docstring and renamed the function to update_text to make it more unique.

Chrisg2000 commented:
I have now implemented tests.
Under Windows CI also a "real test", which should work on a user computer. Only in case the Windows CI Server has special properties regarding the GUI, the test is covered by xfail.

I have also added some documentation.

Chrisg2000 commented on bootloader/src/pyi_splash.c:
```diff
@@ -42,6 +48,571 @@ typedef struct _ipc_message_head {
+     * so the alpha value for all pixels with color information must be set manually.
+     */
+    for (pos = 0; pos < bitmap_header->biWidth * bitmap_header->biHeight; pos++) {
+        if (pPixels[pos] > 0x00000000) {
```
There seems to be a bug here. One cannot draw text in the color 0x000000 (=black), since the this condition is not met

Chrisg2000 commented on bootloader/src/pyi_splash.c:
```diff
@@ -42,6 +48,571 @@ typedef struct _ipc_message_head {
+
+        /* If the message carries more data, load it into text_buffer */
+        if (msg_buffer.text_length != 0 && GetLastError() == ERROR_MORE_DATA) {
+            if (text_buffer != NULL) {
```
Bug: If the user sends a empty message (meaning updating the text on the splash screen with an empty string) the old text inside the text_buffer is drawn again, since it is not freed.

Chrisg2000 commented on PyInstaller/building/splash.py:
```diff
@@ -0,0 +1,308 @@
+    """
+    Converts a RGB value specified by 0xRRGGBB into 0xBBGGRR.
+    """
+    if not 0x00 <= rgb <= 0xFFFFFF:
```
This should be extracted into `Splash`, because the rgb color in hex is always between `0x00` and `0xFFFFFF`, not only on Windows

Legorooj commented:
@bjones1 @htgoebel

bjones1 commented:
Great addition, but I lack the time to provide a good review.

Legorooj commented:
@htgoebel then.

Legorooj commented:
@Chrisg2000 please fix the lint errors. (See the fails on Travis)

Chrisg2000 commented:
Done! Any messages left in there were actually in there on purpose. So were

- the messages W503 (`line break before binary operator`) are actually irrelevant, since this message is [actually thrown incorrectly](https://www.flake8rules.com/rules/W503.html). 
- I ignored the message F401 (`'pyi_splash' imported but unused`) because of the check where the import is located, since this [cannot lead to an error](https://github.com/PyCQA/pyflakes/issues/297)
- I ignored the message E402 (`module level import not at top of file`), because it was given by the general structure of the file

So unfortunately I had no choice but to remove all messages with `noqa`.

Legorooj commented:
@Chrisg2000 this looks ok now. I'm fine with those exception, and Hartmut should be too.

Chrisg2000 commented:
@Legorooj I saw that you linked this pull-request to #4354 for closing. I thought about that, too, but I decided against it, because:

1. this pull request only implements a splash screen for Windows, not for other operating systems
2. I added #4354 as a [link](https://github.com/pyinstaller/pyinstaller/pull/4581/commits/0753a432759831bf10b09c379c7611e04440854f#diff-665ffb7d19093ce3b903440e23eb6e98R213) in the documentation (like it was a long time for the MERGE feature) because i thought it could be kept as a discussion thread for the splash screen feature.

Of course it's up to you if you want to keep #4354 as a "feature" discussion thread or if you want to close it. If you want to close the thread, the link in the documentation would probably have to be removed.

PS: During the first review htgoebel created a [comment](https://github.com/pyinstaller/pyinstaller/pull/4581#discussion_r363041324) which still marks this pull request as "Changes requested". But I cannot mark this comment as "Resolved". Did I do something wrong or is it a bug?

Legorooj commented:
@Chrisg2000 I've unlinked it; I didn't think of that. I'll have a look at the review, and resolve the ones that need it.

Chrisg2000 commented:
I am convertig this PR into a draft because I got a better idea on how to implement a splash screen for the bootloader and I don't want this PR to be reviewed or merged while I am working on a new one.

The new implementation would deal with the current limitations this PR has (platform-dependent, not easily extendable)

I am thinking about using Tcl/Tk for the splash screen (the same library tkinter uses). Tcl/Tk is bundled with (most) python distributions and provides a platform-independent GUI-Kit, which is easy to use from within C. PyInstaller would only need to bundle a few files extra (on Windows two, Mac and Linux needs investigation) to show the splash screen. Since users who want to have a splash screen probably don't really care about distribution size, wouldn't mind ~3 MB extra. Those files can be extracted in less than 0.1s, so no worries on the splash screen showing up to late. The bootloader would then link dynamically to those libraries like it does to the python library, so no new build dependencies.

Legorooj commented:
@Chrisg2000 good idea! However, if you're starting the work from scratch, then do; new branch, new PR etc.

htgoebel commented:
@Chrisg2000 I was just about reviewing and possible merging this pull-request. I'm looking forward for you new proposal them.

Legorooj commented:
@htgoebel @Chrisg2000 problem: Tk isn't so easy from C it appears. We might have to find another solution.

Chrisg2000 commented:
@Legorooj Why do you think Tcl/Tk isn't easy? Although I just only had time to play a little with Tcl/Tk in C, I find it somewhat acceptable (easier than working around Windows API limitations). According to [wikipedia](https://en.wikipedia.org/wiki/Tcl) and [Tcl-lang wiki](https://wiki.tcl-lang.org/page/How+to+embed+Tcl+in+C+applications) Tcl was designed to be easily embeddable. The only thing I am somewhat concerned about is threading Tcl/Tk, but there are many reference implementations (like _tkinter) which dealt with that in the past too. (I will start actively working on the splashsceen on Tuesday, after my written finals are over)

Legorooj commented:
@Chrish2000 you've found stuff I haven't. I only did a quick search I suppose.

Chrisg2000 commented:
Closing this PR, since #4887 is on its way to completion

llamafilm commented:
I like this feature, and I think a console window splash would be more appropriate than a GUI if you are building a console-only application.  Could we keep both options?  I tried to build this from the `Chrisg2000:splash-feature repo` to test but the build fails.

```
python ./waf distclean all -v
'distclean' finished successfully (0.010s)
'all' finished successfully (0.000s)
'distclean' finished successfully (0.000s)
Setting top to                           : C:\Users\Colorfront\Downloads\pyinstaller-splash-feature\pyinstaller-splash-feature\bootloader
Setting out to                           : C:\Users\Colorfront\Downloads\pyinstaller-splash-feature\pyinstaller-splash-feature\bootloader\build
Python Version                           : 3.7.9 (tags/v3.7.9:13c94747c7, Aug 17 2020, 16:30:00) [MSC v.1900 64 bit (AMD64)]
Checking for 'msvc' (C compiler)         : 12:22:30 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\Installer\\vswhere.exe', '-products', '*', '-legacy', '-format', 'json']
12:22:30 runner ['cmd.exe', '/E:on', '/V:on', '/C', 'C:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\waf-print-msvc-1.bat']
12:22:31 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/help']
12:22:31 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/help']
C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.27.29110\bin\HostX64\x64\CL.exe
Checking size of pointer                 : 12:22:31 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '..\\test.c', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\conf_check_303e219cf805e8bf7cae31152a34da6b\\testbuild\\test.c.1.o']
12:22:31 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '..\\test.c', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\conf_check_df399ceca844fdee226fb77c6f4a4073\\testbuild\\test.c.1.o']
12:22:31 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\LINK.exe', '/nologo', '/MANIFEST', 'test.c.1.o', '/OUT:C:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\conf_check_df399ceca844fdee226fb77c6f4a4073\\testbuild\\testprog.exe', '/LIBPATH:C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\lib\\x64', '/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\lib\\10.0.18362.0\\ucrt\\x64', '/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\lib\\10.0.18362.0\\um\\x64', '/LIBPATH:C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\lib\\x64', '/LIBPATH:C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\lib\\x86\\store\\references', '/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\UnionMetadata\\10.0.18362.0', '/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\References\\10.0.18362.0', '/LIBPATH:C:\\Windows\\Microsoft.NET\\Framework64\\v4.0.30319']
12:22:31 runner ['C:\\Program Files (x86)\\Windows Kits\\10\\bin\\10.0.18362.0\\x64\\MT.exe', '/nologo', '-manifest', 'C:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\conf_check_df399ceca844fdee226fb77c6f4a4073\\testbuild\\testprog.exe.manifest', '-outputresource:C:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\conf_check_df399ceca844fdee226fb77c6f4a4073\\testbuild\\testprog.exe;1']
8
Platform                                 : Windows-64bit detected based on compiler
Checking for function unsetenv           : 12:22:32 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '..\\test.cpp', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\conf_check_2b863466c990906d29ccc349c383bbe8\\testbuild\\test.cpp.1.o']
no
Checking for function mkdtemp            : 12:22:32 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '..\\test.cpp', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\conf_check_35bda8a5d479c0fc31bb3ca93a5c8147\\testbuild\\test.cpp.1.o']
no
Checking for function strndup            : 12:22:32 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '..\\test.cpp', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\conf_check_0dd60a7756386d5da02cc6ba31c28717\\testbuild\\test.cpp.1.o']
no
Checking for function strnlen            : 12:22:32 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '..\\test.cpp', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\conf_check_21240939c9ddbcc20d9e6d66cf882183\\testbuild\\test.cpp.1.o']
no
'configure' finished successfully (2.122s)
'make_all' finished successfully (0.016s)
Waf: Entering directory `C:\Users\Colorfront\Downloads\pyinstaller-splash-feature\pyinstaller-splash-feature\bootloader\build\debug'
[ 1/22] Compiling zlib\unzip.c
12:22:32 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/W3', '/EHa', '-O2', '-DUNICODE', '-D_UNICODE', '/Izlib', '/I..\\..\\zlib', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '/DLAUNCH_DEBUG', '/DNDEBUG', '..\\..\\zlib\\unzip.c', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\debug\\zlib\\unzip.c.1.o']
[ 2/22] Compiling zlib\inffast.c
12:22:32 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/W3', '/EHa', '-O2', '-DUNICODE', '-D_UNICODE', '/Izlib', '/I..\\..\\zlib', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '/DLAUNCH_DEBUG', '/DNDEBUG', '..\\..\\zlib\\inffast.c', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\debug\\zlib\\inffast.c.1.o']
[ 3/22] Compiling zlib\adler32.c
12:22:32 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/W3', '/EHa', '-O2', '-DUNICODE', '-D_UNICODE', '/Izlib', '/I..\\..\\zlib', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '/DLAUNCH_DEBUG', '/DNDEBUG', '..\\..\\zlib\\adler32.c', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\debug\\zlib\\adler32.c.1.o']
[ 4/22] Compiling zlib\ioapi.c
12:22:32 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/W3', '/EHa', '-O2', '-DUNICODE', '-D_UNICODE', '/Izlib', '/I..\\..\\zlib', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '/DLAUNCH_DEBUG', '/DNDEBUG', '..\\..\\zlib\\ioapi.c', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\debug\\zlib\\ioapi.c.1.o']
[ 5/22] Compiling zlib\inftrees.c
12:22:32 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/W3', '/EHa', '-O2', '-DUNICODE', '-D_UNICODE', '/Izlib', '/I..\\..\\zlib', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '/DLAUNCH_DEBUG', '/DNDEBUG', '..\\..\\zlib\\inftrees.c', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\debug\\zlib\\inftrees.c.1.o']
[ 6/22] Compiling zlib\crc32.c
12:22:32 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/W3', '/EHa', '-O2', '-DUNICODE', '-D_UNICODE', '/Izlib', '/I..\\..\\zlib', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '/DLAUNCH_DEBUG', '/DNDEBUG', '..\\..\\zlib\\crc32.c', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\debug\\zlib\\crc32.c.1.o']
[ 7/22] Compiling zlib\zutil.c
12:22:32 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/W3', '/EHa', '-O2', '-DUNICODE', '-D_UNICODE', '/Izlib', '/I..\\..\\zlib', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '/DLAUNCH_DEBUG', '/DNDEBUG', '..\\..\\zlib\\zutil.c', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\debug\\zlib\\zutil.c.1.o']
[ 8/22] Compiling zlib\inflate.c
12:22:32 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/W3', '/EHa', '-O2', '-DUNICODE', '-D_UNICODE', '/Izlib', '/I..\\..\\zlib', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '/DLAUNCH_DEBUG', '/DNDEBUG', '..\\..\\zlib\\inflate.c', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\debug\\zlib\\inflate.c.1.o']
[ 9/22] Compiling src\pyi_python.c
12:22:32 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/W3', '/EHa', '-O2', '-DUNICODE', '-D_UNICODE', '/Isrc', '/I..\\..\\src', '/Iwindows', '/I..\\..\\windows', '/Izlib', '/I..\\..\\zlib', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '/DLAUNCH_DEBUG', '/DNDEBUG', '..\\..\\src\\pyi_python.c', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\debug\\src\\pyi_python.c.2.o']
[10/22] Compiling src\pyi_archive.c
12:22:32 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/W3', '/EHa', '-O2', '-DUNICODE', '-D_UNICODE', '/Isrc', '/I..\\..\\src', '/Iwindows', '/I..\\..\\windows', '/Izlib', '/I..\\..\\zlib', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '/DLAUNCH_DEBUG', '/DNDEBUG', '..\\..\\src\\pyi_archive.c', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\debug\\src\\pyi_archive.c.2.o']
[11/22] Compiling src\main.c
12:22:32 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/W3', '/EHa', '-O2', '-DUNICODE', '-D_UNICODE', '/Isrc', '/I..\\..\\src', '/Iwindows', '/I..\\..\\windows', '/Izlib', '/I..\\..\\zlib', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '/DLAUNCH_DEBUG', '/DNDEBUG', '..\\..\\src\\main.c', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\debug\\src\\main.c.2.o']
[12/22] Compiling src\pyi_launch.c
12:22:32 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/W3', '/EHa', '-O2', '-DUNICODE', '-D_UNICODE', '/Isrc', '/I..\\..\\src', '/Iwindows', '/I..\\..\\windows', '/Izlib', '/I..\\..\\zlib', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '/DLAUNCH_DEBUG', '/DNDEBUG', '..\\..\\src\\pyi_launch.c', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\debug\\src\\pyi_launch.c.2.o']
[13/22] Compiling src\pyi_path.c
12:22:32 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/W3', '/EHa', '-O2', '-DUNICODE', '-D_UNICODE', '/Isrc', '/I..\\..\\src', '/Iwindows', '/I..\\..\\windows', '/Izlib', '/I..\\..\\zlib', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '/DLAUNCH_DEBUG', '/DNDEBUG', '..\\..\\src\\pyi_path.c', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\debug\\src\\pyi_path.c.2.o']
inffast.c

ioapi.c
C:\Users\Colorfront\Downloads\pyinstaller-splash-feature\pyinstaller-splash-feature\bootloader\zlib\ioapi.c(85): warning C4996: 'fopen': This function or variable may be unsafe. Consider using fopen_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.

zutil.c

unzip.c

inftrees.c

crc32.c

adler32.c

inflate.c

pyi_python.c

main.c

pyi_path.c
C:\Users\Colorfront\Downloads\pyinstaller-splash-feature\pyinstaller-splash-feature\bootloader\src\pyi_path.c(279): warning C4101: 'buffer': unreferenced local variable

pyi_launch.c

pyi_archive.c
C:\Users\Colorfront\Downloads\pyinstaller-splash-feature\pyinstaller-splash-feature\bootloader\src\pyi_archive.c(273): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data

[14/22] Compiling windows\run.rc
12:22:33 runner ['C:\\Program Files (x86)\\Windows Kits\\10\\bin\\10.0.18362.0\\x64\\RC.exe', '/Isrc', '/I..\\..\\src', '/Iwindows', '/I..\\..\\windows', '/Izlib', '/I..\\..\\zlib', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '/DLAUNCH_DEBUG', '/DNDEBUG', '/fo', 'windows\\run.res', '..\\..\\windows\\run.rc']
[15/22] Compiling src\pyi_utils.c
12:22:33 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/W3', '/EHa', '-O2', '-DUNICODE', '-D_UNICODE', '/Isrc', '/I..\\..\\src', '/Iwindows', '/I..\\..\\windows', '/Izlib', '/I..\\..\\zlib', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '/DLAUNCH_DEBUG', '/DNDEBUG', '..\\..\\src\\pyi_utils.c', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\debug\\src\\pyi_utils.c.2.o']
[16/22] Compiling src\pyi_main.c
12:22:33 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/W3', '/EHa', '-O2', '-DUNICODE', '-D_UNICODE', '/Isrc', '/I..\\..\\src', '/Iwindows', '/I..\\..\\windows', '/Izlib', '/I..\\..\\zlib', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '/DLAUNCH_DEBUG', '/DNDEBUG', '..\\..\\src\\pyi_main.c', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\debug\\src\\pyi_main.c.2.o']
[17/22] Compiling src\pyi_pythonlib.c
12:22:33 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/W3', '/EHa', '-O2', '-DUNICODE', '-D_UNICODE', '/Isrc', '/I..\\..\\src', '/Iwindows', '/I..\\..\\windows', '/Izlib', '/I..\\..\\zlib', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '/DLAUNCH_DEBUG', '/DNDEBUG', '..\\..\\src\\pyi_pythonlib.c', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\debug\\src\\pyi_pythonlib.c.2.o']
[18/22] Compiling src\pyi_splash.c
12:22:33 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/W3', '/EHa', '-O2', '-DUNICODE', '-D_UNICODE', '/Isrc', '/I..\\..\\src', '/Iwindows', '/I..\\..\\windows', '/Izlib', '/I..\\..\\zlib', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '/DLAUNCH_DEBUG', '/DNDEBUG', '..\\..\\src\\pyi_splash.c', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\debug\\src\\pyi_splash.c.2.o']
[19/22] Compiling src\pyi_global.c
12:22:33 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/W3', '/EHa', '-O2', '-DUNICODE', '-D_UNICODE', '/Isrc', '/I..\\..\\src', '/Iwindows', '/I..\\..\\windows', '/Izlib', '/I..\\..\\zlib', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '/DLAUNCH_DEBUG', '/DNDEBUG', '..\\..\\src\\pyi_global.c', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\debug\\src\\pyi_global.c.2.o']
[20/22] Compiling src\pyi_win32_utils.c
12:22:33 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\CL.exe', '/nologo', '/W3', '/EHa', '-O2', '-DUNICODE', '-D_UNICODE', '/Isrc', '/I..\\..\\src', '/Iwindows', '/I..\\..\\windows', '/Izlib', '/I..\\..\\zlib', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\include', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt', '/I..\\..\\..\\..\\..\\..\\..\\..\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt', '/DWIN32', '/DLAUNCH_DEBUG', '/DNDEBUG', '..\\..\\src\\pyi_win32_utils.c', '/FC', '/c', '/FoC:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\debug\\src\\pyi_win32_utils.c.2.o']
Microsoft (R) Windows (R) Resource Compiler Version 10.0.10011.16384
Copyright (C) Microsoft Corporation.  All rights reserved.


[21/22] Linking build\debug\static_zlib.lib
12:22:33 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\LIB.exe', '/nologo', '/OUT:static_zlib.lib', 'zlib\\adler32.c.1.o', 'zlib\\crc32.c.1.o', 'zlib\\inffast.c.1.o', 'zlib\\inflate.c.1.o', 'zlib\\inftrees.c.1.o', 'zlib\\ioapi.c.1.o', 'zlib\\unzip.c.1.o', 'zlib\\zutil.c.1.o']
pyi_utils.c

pyi_main.c

pyi_win32_utils.c

pyi_global.c

pyi_pythonlib.c
C:\Users\Colorfront\Downloads\pyinstaller-splash-feature\pyinstaller-splash-feature\bootloader\src\pyi_pythonlib.c(59): warning C4101: 'p': unreferenced local variable

pyi_splash.c

[22/22] Linking build\debug\run_d.exe
12:22:34 runner ['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\LINK.exe', '/nologo', '/MANIFEST', '/MACHINE:X64', '/SUBSYSTEM:CONSOLE,5.02', 'src\\main.c.2.o', 'src\\pyi_archive.c.2.o', 'src\\pyi_global.c.2.o', 'src\\pyi_launch.c.2.o', 'src\\pyi_main.c.2.o', 'src\\pyi_path.c.2.o', 'src\\pyi_python.c.2.o', 'src\\pyi_pythonlib.c.2.o', 'src\\pyi_splash.c.2.o', 'src\\pyi_utils.c.2.o', 'src\\pyi_win32_utils.c.2.o', 'windows\\run.res', '/OUT:C:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\debug\\run_d.exe', '/LIBPATH:.', 'static_zlib.lib', '/LIBPATH:C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\lib\\x64', '/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\lib\\10.0.18362.0\\ucrt\\x64', '/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\lib\\10.0.18362.0\\um\\x64', '/LIBPATH:C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\lib\\x64', '/LIBPATH:C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\lib\\x86\\store\\references', '/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\UnionMetadata\\10.0.18362.0', '/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\References\\10.0.18362.0', '/LIBPATH:C:\\Windows\\Microsoft.NET\\Framework64\\v4.0.30319', 'user32.lib', 'comctl32.lib', 'kernel32.lib', 'advapi32.lib', 'ws2_32.lib', 'gdi32.lib']
libucrt.lib(strnlen.obj) : error LNK2005: strnlen already defined in pyi_utils.c.2.o
C:\Users\Colorfront\Downloads\pyinstaller-splash-feature\pyinstaller-splash-feature\bootloader\build\debug\run_d.exe : fatal error LNK1169: one or more multiply defined symbols found

Waf: Leaving directory `C:\Users\Colorfront\Downloads\pyinstaller-splash-feature\pyinstaller-splash-feature\bootloader\build\debug'
Build failed
 -> task in 'run_d' failed with exit status 1169:
        {task 1688776105656: cprogram main.c.2.o,pyi_archive.c.2.o,pyi_global.c.2.o,pyi_launch.c.2.o,pyi_main.c.2.o,pyi_path.c.2.o,pyi_python.c.2.o,pyi_pythonlib.c.2.o,pyi_splash.c.2.o,pyi_utils.c.2.o,pyi_win32_utils.c.2.o,run.res -> run_d.exe}
['C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\bin\\HostX64\\x64\\LINK.exe', '/nologo', '/MANIFEST', '/MACHINE:X64', '/SUBSYSTEM:CONSOLE,5.02', 'src\\main.c.2.o', 'src\\pyi_archive.c.2.o', 'src\\pyi_global.c.2.o', 'src\\pyi_launch.c.2.o', 'src\\pyi_main.c.2.o', 'src\\pyi_path.c.2.o', 'src\\pyi_python.c.2.o', 'src\\pyi_pythonlib.c.2.o', 'src\\pyi_splash.c.2.o', 'src\\pyi_utils.c.2.o', 'src\\pyi_win32_utils.c.2.o', 'windows\\run.res', '/OUT:C:\\Users\\Colorfront\\Downloads\\pyinstaller-splash-feature\\pyinstaller-splash-feature\\bootloader\\build\\debug\\run_d.exe', '/LIBPATH:.', 'static_zlib.lib', '/LIBPATH:C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\lib\\x64', '/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\lib\\10.0.18362.0\\ucrt\\x64', '/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\lib\\10.0.18362.0\\um\\x64', '/LIBPATH:C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\lib\\x64', '/LIBPATH:C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.27.29110\\lib\\x86\\store\\references', '/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\UnionMetadata\\10.0.18362.0', '/LIBPATH:C:\\Program Files (x86)\\Windows Kits\\10\\References\\10.0.18362.0', '/LIBPATH:C:\\Windows\\Microsoft.NET\\Framework64\\v4.0.30319', 'user32.lib', 'comctl32.lib', 'kernel32.lib', 'advapi32.lib', 'ws2_32.lib', 'gdi32.lib']
PS C:\Users\Colorfront\Downloads\pyinstaller-splash-feature\pyinstaller-splash-feature\bootloader>
```
